crates/testing/tests/combined_network.rs

```diff
@@ -33,34 +76,219 @@ async fn test_combined_network() {
         // allow more time to pass in CI
         completion_task_description: CompletionTaskDescription::TimeBasedCompletionTaskBuilder(
             TimeBasedCompletionTaskDescription {
-                duration: Duration::from_millis(1_200_000),
+                duration: Duration::from_secs(120),
             },
         ),
         ..TestMetadata::default_multiple_rounds()
     };
 
+    let mut all_nodes = vec![];
+    for node in 0..metadata.total_nodes {
+        all_nodes.push(ChangeNode {
+            idx: node,
+            updown: UpDown::NetworkDown,
+        });
+    }
+
+    metadata.spinning_properties = SpinningTaskDescription {
+        node_changes: vec![(Duration::new(4, 0), all_nodes)],
+    };
+
     metadata
         .gen_launcher::<TestTypes, CombinedImpl>(0)
         .launch()
         .run_test()
-        .await
+        .await;
+
+    // async_std needs time to spin down the handler
+    #[cfg(async_executor_impl = "async-std")]
+    async_std::task::sleep(Duration::from_secs(4)).await;
 }
 
-// stress test for web server with libp2p
+// A run where the webserver crashes partway through
+// and then comes back up
+#[cfg_attr(
+    async_executor_impl = "tokio",
+    tokio::test(flavor = "multi_thread", worker_threads = 2)
+)]
+#[cfg_attr(async_executor_impl = "async-std", async_std::test)]
+#[instrument]
+async fn test_combined_network_reup() {
```

Comment:
To clarify my understanding, what this test is doing is:
1)  spins all nodes up up
2) waits for 6 seconds,
3) spins all the networks down. 
4) waits 6 seconds and spins all the networks back up

A lingering concern: it doesn't look like the safety task (overall_safety_task.rs) looks at times the entire network is down. That is, it looks only at the overall # views completed. But, I think the invariant to express here is "we make progress after step 4". So, it seems plausible to me that the test could succeed even if no progress is made after step 4. WDYT? Maybe I'm missing something

As far as a fix (assuming this is a problem, and definitely out of scope for this PR): I wonder if we can insert a hook to have a custom property record the #successful views after step (3), and, again when the task completes. If their difference is positive, we pass, else we fail. The testing infra should have the ability to do this with a custom hook

Reply:
Another possible fix is to change it to allow `UpDown` to work as a function of rounds rather than a function of time - WDYT?

Reply:
Resolving in favor of the separate issue #2094

Reply:
+1 +1